keps/sig-network/2079-network-policy-port-range/README.md

```diff
@@ -232,20 +232,42 @@ _This section must be completed when targeting alpha to a release._
 
 * **Can the feature be disabled once it has been enabled (i.e. can we roll back
   the enablement)?**
-  Yes, but CNIs relying on the new field wont recognize it anymore
+  Yes, but disabling the feature gate and updating a Network Policy with existing 
```

Comment:
It should not remove the field - gates should be coded to preserve the contents of the field if it is exists, even if the feature itself is disabled.

Reply:
Is this basically a difference in storage vs. what is presented via the API? Seems like we should not remove the end port field from stored data, but depending on the gate we would either present the field or hide it?

Definitely want to maintain the behavior where disabling the gate -> enabling the gate -> disabling the gate leaves you back where you started.

Reply:
Hum, ok, seems reasonable. Just need in the impl (https://github.com/kubernetes/kubernetes/pull/97058/files#diff-f0064039f0ac16d6f72b07c2afe0fd12232dae1300dda4e3f64c72209666324bR102) to change this behavior to only drop IF the old netpol already does not have the field :)

Will change here the KEP and then the impl :)

About what's shown in the API I honestly don't know if it's possible to persist the data but not show it, maybe Tim knows

Reply:
Here a quick question: what if the field is existing on the old object, and I only change the value? I'm ok about keeping the field or allowing the user to drop it, but what's the expected behavior when FG is disabled and the user changes only the contents?

Reply:
The "and fieldNotUsed" pattern is what we use in this case, see e.g. this one as an example:
https://github.com/kubernetes/kubernetes/blob/f6e04cd3ad1f3a1a00fd091e1524d2d27dc56bd7/pkg/api/pod/util.go#L442

Reply:
Oh thank you! saved me a lot of time! :)